### PR TITLE
To-Do List Project 2

### PR DESCRIPTION
**Pull request by david-lafontant**
**To-Do List Project 2**

- Removed all hardcoded items from the tasks array.
- Created a new JavaScript file for the new functionality.
- implement a function for adding a new task.
- implement a function for deleting a task.
- implement a function for editing task descriptions.
- By default new tasks have the property completed set to false and the property index set to the value of the new array length.
- Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique.
- All changes to the To Do List saved in local storage.

### CODE REVIEW

File: src/modules/task.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+export default class Task {
+  constructor(description, id, completed) {
+    this.description = description;
+    this.id = id;
```
Comment (brytebee) [inline src/modules/task.js:4]: - [ ] **Required:** Your list id's should begin from 1. At the moment they begin from 0. When items are added to the list and the page is refreshed, the ids begin from zero again. Let's fix these.
I have added a demo gif to explain the requested changes.

fix id

**Review by brytebee — CHANGES_REQUESTED** [review]
Hi, @david-lafontant👋,

Good job so far coming so far with the progress you've made! 💪💪

## Highlights
You have done amazing in the following:
- an excellent description of PR✔️
- clean code ✔️
- well written README✔️
 
However, there are some issues that you still need to work on to go to the next project but you are almost there!



# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/task.js (hunk 1 above)
Comment (david-lafontant) [reply src/modules/task.js:4]: @brytebee thank you for your comment. I think I fixed it.

**Review by andersonlebon — APPROVED** [review]
Hi @david-lafontant 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 :clap: :clap: :100: 

nodding-yes

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- You  are following GitHub actions :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Your README file is now professional :heavy_check_mark: 

## Optional suggestions
- [ ] Kindly Check the comments under the review._
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by david-lafontant** [comment]
> Hi @david-lafontant Your project is complete! There is nothing else to say other than... it's time to merge it :shipit: Congratulations! 🎉 👏 👏 💯
> 
> nodding-yes
> 
> ## To highlight 🍾
> * All linters are passing ✔️
> * You  are following GitHub actions ✔️
> * Good commit history ✔️
> * Good PR description ✔️
> * Your README file is now professional ✔️
> 
> ## Optional suggestions
> * [ ]  Kindly Check the comments under the review._
>   _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
> 
> Cheers and Happy coding!👏👏👏
> 
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
> 
> _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by EricMbouwe — COMMENTED** [review]
### Hi @david-lafontant, 

# Status: Approved ✔️
Excellent job on this project. Keep it up 👏
Your project is complete! There is nothing else to say other than... it's time to merge it 💪
Congratulations! 🎉

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**